File: app/src/main/java/com/example/unsplash/model/models/AccessToken.java
package com.example.unsplash.model.models;


import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

/**
 * Access token.
 * */

public class AccessToken implements Parcelable {

    /**
     * access_token : 091343ce13c8ae780065ecb3b13dc903475dd22cb78a05503c2e0c69c5e98044
     * token_type : bearer
     * scope : public read_photos write_photos
     * created_at : 1436544465
     */
    @SerializedName("access_token")
    private String access_token;
    @SerializedName("token_type")
    private String token_type;
    @SerializedName("scope")
    private String scope;
    @SerializedName("created_at")
    private int created_at;

    public String getAccess_token() {
        return access_token;
    }

    public String getToken_type() {
        return token_type;
    }

    public String getScope() {
        return scope;
    }

    public int getCreated_at() {
        return created_at;
    }

    private AccessToken(Parcel in) {
        access_token = in.readString();
        token_type = in.readString();
        scope = in.readString();
        created_at = in.readInt();
    }

    public static final Creator<AccessToken> CREATOR = new Creator<AccessToken>() {
        @Override
        public AccessToken createFromParcel(Parcel in) {
            return new AccessToken(in);
        }

        @Override
        public AccessToken[] newArray(int size) {
            return new AccessToken[size];
        }
    };

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(Parcel dest, int flags) {
        dest.writeString(access_token);
        dest.writeString(token_type);
        dest.writeString(scope);
        dest.writeInt(created_at);
    }
}

File: app/src/main/java/com/example/unsplash/model/models/AccessToken.kt
package com.example.unsplash.model.models

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.SerializedName

/**
 * Access token.
 */

class AccessToken private constructor(`in`: Parcel) : Parcelable {

    /**
     * accessToken : 091343ce13c8ae780065ecb3b13dc903475dd22cb78a05503c2e0c69c5e98044
     * tokenType : bearer
     * scope : public read_photos write_photos
     * createdAt : 1436544465
     */
    @SerializedName("access_token")
    val accessToken: String? = `in`.readString()
    @SerializedName("token_type")
    val tokenType: String? = `in`.readString()
    @SerializedName("scope")
    val scope: String? = `in`.readString()
    @SerializedName("created_at")
    val createdAt: Int = `in`.readInt()

    override fun describeContents(): Int {
        return 0
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeString(accessToken)
        dest.writeString(tokenType)
        dest.writeString(scope)
        dest.writeInt(createdAt)
    }

    companion object CREATOR : Parcelable.Creator<AccessToken> {
        override fun createFromParcel(parcel: Parcel): AccessToken {
            return AccessToken(parcel)
        }

        override fun newArray(size: Int): Array<AccessToken?> {
            return arrayOfNulls(size)
        }
    }
}
